frontend/rpm/template.go

```diff
@@ -201,13 +201,17 @@ func (w *specWrapper) PrepareSources() (fmt.Stringer, error) {
 	for _, name := range keys {
 		src := w.Spec.Sources[name]
 		err := func(name string, src dalec.Source) error {
-			if patches[name] {
+			isDir := dalec.SourceIsDir(src)
```

Comment:
Instead of handing the "maybe it is a file in a dir" down below, it would be nice if `dalec.SourceIsDir` returned the actual truth.

To do this, we'll need to modify the `SourceOpt` type to include a field for interacting with the filesystem on an `llb.State`.
After we spoke about this last week I was even thinking we could provide an implementation of https://pkg.go.dev/io/fs#ReadDirFS for use in `SourceIsDir` and potentially other places.

Reply:
I really like the `ReadDirFS` idea, and I opened an issue: #192